crml/erc20-peg/src/tests.rs

```diff
@@ -0,0 +1,163 @@
+use super::*;
+use crate::mock::{DepositEventSignature, Erc20Peg, ExtBuilder, GenericAsset, PegPalletId, Test};
+use cennznet_primitives::types::AccountId;
+use crml_support::MultiCurrency;
+use frame_support::assert_ok;
+use hex_literal::hex;
+
+#[test]
+fn deposit_claim() {
+	ExtBuilder::default().build().execute_with(|| {
+		assert_ok!(Erc20Peg::activate_deposits(frame_system::RawOrigin::Root.into(), true));
+		let origin: AccountId =
+			AccountId::from(hex!("0000000000000000000000a86e122edbdcba4bf24a2abf89f5c230b37df49d4a"));
+		let token_address: H160 = H160::default();
+		let amount: Balance = 100;
+		let beneficiary: H256 = H256::default();
+		let claim = Erc20DepositEvent {
+			token_address,
+			amount: amount.into(),
+			beneficiary,
+		};
+		let tx_hash = H256::default();
+
+		assert_ok!(Erc20Peg::deposit_claim(Some(origin).into(), tx_hash, claim));
+	});
+}
+
+#[test]
+fn on_deposit_mints() {
+	ExtBuilder::default().build().execute_with(|| {
+		let contract_address: EthAddress = EthAddress::from(Erc20Peg::contract_address());
+		assert_ok!(Erc20Peg::set_contract_address(
+			frame_system::RawOrigin::Root.into(),
+			contract_address
+		));
+
+		let token_address: H160 = H160::default();
+		let amount: Balance = 100;
+		let beneficiary: H256 = H256::default();
+		let claim = Erc20DepositEvent {
+			token_address,
+			amount: amount.into(),
+			beneficiary,
+		};
+		let event_claim_id: u64 = 0;
+		let event_type: H256 = DepositEventSignature::get().into();
+		Erc20Peg::on_success(event_claim_id, &contract_address, &event_type, &claim.encode());
+
+		let beneficiary: AccountId = AccountId::decode(&mut &beneficiary.0[..]).unwrap();
+		let expected_asset_id = 0;
```

Comment:
its unusual for the asset minted to have asset_id 0 in practice since cennz and cpay would exist already.
check the generic asset IDs are set in the mock storage